console.log(powerOfTwo(3)); // вызов классической функции

/**
 * Обычная функция, объявленная единожды может быть вызвана неоднократно
 * Может быть вызвана выше чем она объявлена - это называется "всплывание"
 * @param {*} number
 * @returns
 */
function powerOfTwo(number) {
  return number * number;
}

/*
 * вызов анонимной функции вышеобъявления
 * (здесь вызов приведет к ошибке так как переменная объявлена ниже и ней еще ничего не известно)
 */
// console.log(pwroftwo(4));

/**
 * Анонимная функция кладется в переменную и не может быть вызвана выше обхявления переменной
 * Это одна из ключевых особенностей(отличий) от обычных функций
 *
 * @param {*} number
 * @returns
 */
const pwroftwo = function (number) {
  return number * number;
};

console.log(pwroftwo(4)); // вызов анонимной функции (здесь она отработает успешно)
